### ISSUE TITLE
Pagecache doesn't compile on stable rust

### ISSUE THREAD

**Issue by codri** [spacejam/sled#290]
1. Sled Version: 0.15.11
2. Rustc Version: rustc 1.24.1 (d3ae9a9e0 2018-02-27)
3. OS: Windows 10

The following code doesn't compile on latest stable rust, neither on nightly.

Cargo.toml
```toml
[dependencies]
log = "0.4"
env_logger = "0.5"
sled = "0.15.11"
```

```rs
extern crate sled;

use sled::{ConfigBuilder, Tree};

fn main() {
    println!("Hello, world!");
}
```

errors

```sh
error[E0599]: no method named `read_message` found for type `std::sync::Arc<std::fs::File>` in the current scope
  --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\io\iobuf.rs:85:28
   |
85 |                 match file.read_message(
   |                            ^^^^^^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `read_message`, perhaps you need to implement it:
           candidate #1: `io::reader::LogReader`

error[E0599]: no method named `read_message` found for type `std::sync::Arc<std::fs::File>` in the current scope
  --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\io\iterator.rs:76:25
   |
76 |                 match f.read_message(
   |                         ^^^^^^^^^^^^
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `read_message`, perhaps you need to implement it:
           candidate #1: `io::reader::LogReader`

error[E0599]: no method named `read_segment_header` found for type `std::sync::Arc<std::fs::File>` in the current scope
   --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\io\iterator.rs:146:32
    |
146 |         let segment_header = f.read_segment_header(offset)?;
    |                                ^^^^^^^^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `read_segment_header`, perhaps you need to implement it:
            candidate #1: `io::reader::LogReader`

error[E0599]: no method named `read_segment_trailer` found for type `std::sync::Arc<std::fs::File>` in the current scope
   --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\io\iterator.rs:186:33
    |
186 |         let segment_trailer = f.read_segment_trailer(trailer_offset);
    |                                 ^^^^^^^^^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `read_segment_trailer`, perhaps you need to implement it:
            candidate #1: `io::reader::LogReader`

error[E0599]: no method named `read_message` found for type `std::sync::Arc<std::fs::File>` in the current scope
   --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\io\log.rs:141:22
    |
141 |         let read = f.read_message(
    |                      ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `read_message`, perhaps you need to implement it:
            candidate #1: `io::reader::LogReader`

error[E0599]: no method named `read_segment_header` found for type `std::sync::Arc<std::fs::File>` in the current scope
    --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\io\segment.rs:1045:31
     |
1045 |     while let Ok(segment) = f.read_segment_header(cursor) {
     |                               ^^^^^^^^^^^^^^^^^^^
     |
     = help: items from traits can only be used if the trait is implemented and in scope
     = note: the following trait defines an item `read_segment_header`, perhaps you need to implement it:
             candidate #1: `io::reader::LogReader`

error[E0599]: no method named `read_segment_trailer` found for type `&std::fs::File` in the current scope
    --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\io\segment.rs:1108:29
     |
1108 |         let trailer_res = f.read_segment_trailer(trailer_lid);
     |                             ^^^^^^^^^^^^^^^^^^^^
     |
     = help: items from traits can only be used if the trait is implemented and in scope
     = note: the following trait defines an item `read_segment_trailer`, perhaps you need to implement it:
             candidate #1: `io::reader::LogReader`

error[E0599]: no method named `read_message` found for type `std::sync::Arc<std::fs::File>` in the current scope
   --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\config.rs:582:19
    |
582 |                 f.read_message(
    |                   ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `read_message`, perhaps you need to implement it:
            candidate #1: `io::reader::LogReader`

error[E0599]: no method named `read_message` found for type `std::sync::Arc<std::fs::File>` in the current scope
   --> C:\Users\user\.cargo\registry\src\github.com-1ecc6299db9ec823\pagecache-0.4.0\src\config.rs:597:19
    |
597 |                 f.read_message(
    |                   ^^^^^^^^^^^^
    |
    = help: items from traits can only be used if the trait is implemented and in scope
    = note: the following trait defines an item `read_message`, perhaps you need to implement it:
            candidate #1: `io::reader::LogReader`

error: aborting due to 9 previous errors

error: Could not compile `pagecache`.
```

**Comment by spacejam** [comment]
Hey, @codri ! Congratulations, you are the first windows user who has raised an issue about this not working yet! I don't often have access to a windows machine, so I'm flying blind here, but I just pushed https://github.com/spacejam/sled/pull/291 which might fix your issue. Could you try to build master again once I merge this PR? I'm eager to get this working for you, but I'm a bit restricted to "mars rover debugging" for the next few days for windows.

**Comment by spacejam** [comment]
published sled `0.15.2` and pagecache `0.4.1` with the above fix. Sled doctests now work on windows. Please don't hesitate to open another issue if something does not meet your expectations! Thanks for opening this issue, @codri !

**Comment by 6D65** [comment]
Thank you, @spacejam , it does compile with the latest version.
Looking forward to play with it.

It's unfortunate that Travis CI doesn't seem to have Windows machines, though.
